compatibility/bazel_tools/daml_trigger/daml_trigger.bzl

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+load(
+    "@daml//bazel_tools/client_server:client_server_test.bzl",
+    "client_server_test",
+)
+load("//bazel_tools:versions.bzl", "version_to_name")
+
+def daml_trigger_dar(sdk_version):
+    daml = "@daml-sdk-{sdk_version}//:daml".format(
+        sdk_version = sdk_version,
+    )
+    native.genrule(
+        name = "trigger-example-dar-{sdk_version}".format(
+            sdk_version = version_to_name(sdk_version),
+        ),
+        srcs = [
+            "//bazel_tools/daml_trigger:example/src/CopyTrigger.daml",
+            "//bazel_tools/daml_trigger:example/src/TestScript.daml",
+        ],
+        outs = ["trigger-example-{sdk_version}.dar".format(
+            sdk_version = version_to_name(sdk_version),
+        )],
+        tools = [daml],
+        cmd = """\
+set -euo pipefail
+TMP_DIR=$$(mktemp -d)
+cleanup() {{ rm -rf $$TMP_DIR; }}
+trap cleanup EXIT
+mkdir -p $$TMP_DIR/src
+cp -L $(location //bazel_tools/daml_trigger:example/src/CopyTrigger.daml) $$TMP_DIR/src/
+cp -L $(location //bazel_tools/daml_trigger:example/src/TestScript.daml) $$TMP_DIR/src/
+cat <<EOF >$$TMP_DIR/daml.yaml
+sdk-version: {sdk_version}
+name: trigger-example
+source: src
+parties:
+  - Alice
+  - Bob
+version: 0.0.1
+dependencies:
+  - daml-prim
+  - daml-script
+  - daml-stdlib
+  - daml-trigger
+sandbox-options:
+  - --wall-clock-time
+EOF
+$(location {daml}) build --project-root=$$TMP_DIR -o $$PWD/$(OUTS)
+""".format(
+            daml = daml,
+            sdk_version = sdk_version,
+        ),
+    )
+
+def daml_trigger_test(compiler_version, runner_version):
+    compiled_dar = "//:trigger-example-dar-{version}".format(
+        version = version_to_name(compiler_version),
+    )
+    daml_runner = "@daml-sdk-{version}//:daml".format(
+        version = runner_version,
+    )
+    name = "daml-trigger-test-compiler-{compiler_version}-runner-{runner_version}".format(
+        compiler_version = version_to_name(compiler_version),
+        runner_version = version_to_name(runner_version),
+    )
+    native.genrule(
+        name = "{}-client-sh".format(name),
+        outs = ["{}-client.sh".format(name)],
+        cmd = """\
+cat >$(OUTS) <<'EOF'
+#!/usr/bin/env bash
+set -euo pipefail
+canonicalize_rlocation() {{
```

Comment:
I wonder if it would make sense to do the normalization in the runfiles library itself. Obviously that doesn’t help us in the short term until it lands but it seems weird that Bazel requires this hack.

Reply:
Yes, agreed, that should be fixed somewhere higher upstream. I'll look into whether there's a ticket already or raise one upstream.